session/session.go

```diff
@@ -1098,7 +1098,7 @@ func (s *session) retry(ctx context.Context, maxCnt uint) (err error) {
 			}
 			_, digest := s.sessionVars.StmtCtx.SQLDigest()
 			s.txn.onStmtStart(digest.String())
-			if err = sessiontxn.GetTxnManager(s).OnStmtStart(ctx); err == nil {
+			if err = sessiontxn.GetTxnManager(s).OnStmtStart(ctx, nil); err == nil {
```

Comment:
Can we find a valid statement object here?

Reply:
Actually, stmt node should have been assigned to txnManager before. So, it will use that stmt node within the method instead of nil.

Reply:
Yes, we can.